### PR TITLE
Add ERC: Splitable utility NFT extension for ERC-721

### PR DESCRIPTION
**Pull request by adityashukla700**
Splitable utility NFT extension for ERC-721

When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-template.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your GitHub username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

File: ERCS/erc-7550.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
+---
+eip: 7550
```
Comment (abcoathup) [inline ERCS/erc-7550.md:2]:
```suggestion
eip: 7567
```
Assigning next sequential EIP/ERC/RIP number.
Numbering changed to sequential from 7500 and is no longer the PR number.  7550 had already been assigned.

Please also update the filename.

Comment (adityashukla700) [reply ERCS/erc-7550.md]: made the required change and renamed file to erc-7567.md

File: ERCS/erc-7567.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+
+## Specification
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+**Overview**
+
+Multiutility token contracts, compliant with the [ERC-721](./eip-721.md) standard, `MAY` implement this extension to provide standard methods for `managing` the `distribution` and `transfer` of `utilities` associated with NFTs.
+
+**Token Utility Management:** Token owners `MAY` assign specific utilities to their NFTs using the `mintUtilityNFT(uint256[] memory utilityIndex)` function. This function mints a new NFT with the specified utility IDs. Existing NFTs can also be augmented with additional utilities using the `mintNewUtilityToNFT(uint256 tokenId, uint256 utilityIndex)` function.
```
Comment (SamWilsn) [inline ERCS/erc-7567.md:28]: Just use "MAY" (no quotes, without backticks) when referencing the RFC 2119 keywords. This applies throughout the document.

File: ERCS/erc-7567.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+**Utility Information Access:**
+The contract provides functions for retrieving information about utilities and their association with NFTs:
+- `getUtilitiesOf(uint256 _tokenID)`: It retrieves the list of utility IDs associated with a specific NFT's tokenID.
+- `allUtilitiesList()`: This function retrieves the complete list of available utilities, including their IDs, names, issuers, and transferability status.
+- `allUtilityIndexList()`: This function returns the list of IDs of all available utilities.
+- `utilityMap(uint256 id)`: Returns the details of a specific utility based on its ID.
+
+**Interface**
+```solidity
+// SPDX-License-Identifier: CC0
```
Comment (SamWilsn) [inline ERCS/erc-7567.md:43]:
```suggestion
// SPDX-License-Identifier: CC0-1.0
```

File: ERCS/erc-7567.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+This standard is an extension of [ERC-721](./eip-721.md). It proposes splitable utility properties to a NFT, which will allow user to have or trade specific utilities instead of whole set as an NFT. It gives creators ability to add new utility functionality to their tokens.
+
+
+## Specification
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+**Overview**
+
+Multiutility token contracts, compliant with the [ERC-721](./eip-721.md) standard, `MAY` implement this extension to provide standard methods for `managing` the `distribution` and `transfer` of `utilities` associated with NFTs.
```
Comment (SamWilsn) [inline ERCS/erc-7567.md:26]: More specifically, only use backticks (\`) for code snippets. Use italics or bold for non-code terms with specific definitions.

File: ERCS/erc-7567.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+* @dev Mint a new utility to an existing NFT.
+* @param tokenId The ID of the existing NFT.
+* @param utilityIndex The index of the utility to be minted to the NFT.
+*/
+
+function mintNewUtilityToNFT(uint256 tokenId, uint256 utilityIndex) external;
+
+/**
+* @dev Get the utilities associated with a specific token.
+* @param \_tokenID The ID of the token to query for utilities.
```
Comment (SamWilsn) [inline ERCS/erc-7567.md:95]: The backslash here is unnecessary.

```suggestion
* @param _tokenID The ID of the token to query for utilities. 
```

File: ERCS/erc-7567.md [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+function mintNewUtilityToNFT(uint256 tokenId, uint256 utilityIndex) external;
+
+/**
+* @dev Get the utilities associated with a specific token.
+* @param \_tokenID The ID of the token to query for utilities. 
+* @return An array of utility indices associated with the token.
+*/
+
+function getUtilitiesOf(uint256 \_tokenID) external view returns (uint256[] memory);
```
Comment (SamWilsn) [inline ERCS/erc-7567.md:99]:
```suggestion
function getUtilitiesOf(uint256 _tokenID) external view returns (uint256[] memory);
```

File: ERCS/erc-7567.md [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+**Utility Administrator:** Administrators of the contract `MAY` add new utilities to the list of available utilities using the `addUtilities(Utility memory utility)` function. This function allows for the `expansion` of the utility ecosystem without explicit creation of NFT and the introduction of new utility types.
+
+**Utility Information Access:**
+The contract provides functions for retrieving information about utilities and their association with NFTs:
+- `getUtilitiesOf(uint256 _tokenID)`: It retrieves the list of utility IDs associated with a specific NFT's tokenID.
+- `allUtilitiesList()`: This function retrieves the complete list of available utilities, including their IDs, names, issuers, and transferability status.
+- `allUtilityIndexList()`: This function returns the list of IDs of all available utilities.
+- `utilityMap(uint256 id)`: Returns the details of a specific utility based on its ID.
+
+**Interface**
```
Comment (SamWilsn) [inline ERCS/erc-7567.md:41]:
```suggestion
### Interface
```

File: ERCS/erc-7567.md [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+**Limited Functionality:**
+Traditional NFTs typically represent ownership of digital assets like images or videos but they lack the capability to include additional functionalities or benefits. The Multi utility NFT standard enables NFTs to serve as a platform that can integrate utilities expanding their potential beyond just being tokens of ownership.
+
+  
+
+**Fragmented Utility Management:**
+Managing utilities linked to NFTs can become burdensome and inefficient. The Multi utility NFT standard consolidates these utilities within one NFT simplifying user management and enhancing the utility experience.
+
+  
+
```
Comment (SamWilsn) [inline ERCS/erc-7567.md:138]:
```suggestion
```

File: ERCS/erc-7567.md [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+## Rationale
+The concept of the Multi NFT standard suggests a method, for incorporating multiple functionalities into a single NFT. This approach aims to tackle issues commonly associated with NFT implementations:
+
+
+**Limited Functionality:**
+Traditional NFTs typically represent ownership of digital assets like images or videos but they lack the capability to include additional functionalities or benefits. The Multi utility NFT standard enables NFTs to serve as a platform that can integrate utilities expanding their potential beyond just being tokens of ownership.
+
+  
+
```
Comment (SamWilsn) [inline ERCS/erc-7567.md:133]:
```suggestion
```

File: ERCS/erc-7567.md [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+Managing utilities linked to NFTs can become burdensome and inefficient. The Multi utility NFT standard consolidates these utilities within one NFT simplifying user management and enhancing the utility experience.
+
+  
+
+**Unstructured Utility Integration:**
+Integrating utilities into NFTs often involves custom code and complex interactions, with contracts making it difficult for developers to create reusable utility components. The Multi utility NFT standard introduces an interface that defines and integrates utilities promoting interoperability and facilitating adoption.
+
+  
+
+
```
Comment (SamWilsn) [inline ERCS/erc-7567.md:144]:
```suggestion
```

File: ERCS/erc-7567.md [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+* @dev Get information about a specific utility based on its ID.
+* @param id The ID of the utility to query.
+* @return A utility struct representing the specified utility.
+*/
+
+function utilityMap(uint256 id) external view returns (Utility memory);
+}
+```
+
+## Rationale
```
Comment (SamWilsn) [inline ERCS/erc-7567.md:125]: A lot of your rationale below justifies the existence of the EIP as a whole. The Rationale section should explain technical choices made within the EIP itself.

> Motivation: "Why do we need a shed?"
> Rationale: "Why did we paint the shed red?"

File: ERCS/erc-7567.md [hunk 12]
```diff
@@ -0,0 +1,155 @@
...
+## Backwards Compatibility
+This standard is compatible with [ERC-721](./eip-721.md) standards.
+
+Existing Upgradedable [ERC-721](./eip-721.md) can upgrade to this standard, enabling locking capability inherently and unlock underlying liquidity features.
+
+## Security Considerations
+There are no security considerations related directly to the implementation of this standard for the contract that manages [ERC-721](./eip-721.md).
+
+## Copyright
+Copyright and related rights waived via CC0.
```
Comment (SamWilsn) [inline ERCS/erc-7567.md:155]: Please use the exact copyright waiver from [the template](https://github.com/ethereum/EIPs/blob/92daa1df9f48c0f1edf945ed516a1d696fadb5fc/eip-template.md?plain=1#L120).

**Review by SamWilsn — CHANGES_REQUESTED** [review]

**Pull request closed without merging**